packages/medusa/src/services/order.ts

```diff
@@ -1408,6 +1499,185 @@ class OrderService extends TransactionBaseService {
     })
   }
 
+  async decorateTotalsNew(
+    order: Order,
+    totalsConfig: TotalsConfig = {}
+  ): Promise<Order> {
+    const manager = this.transactionManager_ ?? this.manager_
+    const totalsNewServiceTx = this.totalsNewService_.withTransaction(manager)
+
+    const calculationContext = await this.totalsService_.getCalculationContext(
+      order
+    )
+    const includeTax = totalsConfig?.force_taxes || order.region.automatic_taxes
+    const orderItems = [...order.items]
+    const orderShippingMethods = [...order.shipping_methods]
+    let useExistingTaxLines = totalsConfig.useExistingTaxLines
+
+    // If we are forced to use the existing tax lines then at least one item must have a tax line
+    if (
+      useExistingTaxLines &&
+      !orderItems.some((item) => item.tax_lines?.length)
+    ) {
+      throw new MedusaError(
+        MedusaError.Types.UNEXPECTED_STATE,
+        "Can't compute order totals with totals config useExistingTaxLines set to true but none of the order items contains any tax lines"
+      )
+    }
+
+    // If we are not forced to use the existing tax lines then fetch the tax lines once and then associate them
+    // to the items and methods copy to avoid fetching them later multiple times and then set useExistingTaxLines to force
+    // the total service to use the tax lines from the items and methods
+    if (!useExistingTaxLines) {
+      const taxLinesMaps = await this.taxProviderService_
+        .withTransaction(manager)
+        .getTaxLinesMap(orderItems, calculationContext)
+
+      orderItems.forEach((item) => {
+        if (item.is_return) {
+          return
+        }
+        item.tax_lines = taxLinesMaps.lineItemsTaxLines[item.id] ?? []
+      })
+      orderShippingMethods.forEach((method) => {
+        method.tax_lines = taxLinesMaps.shippingMethodsTaxLines[method.id] ?? []
+      })
+
+      useExistingTaxLines = true
+    }
```

Comment:
thought: when the order has been placed we have "locked in" so think we should remove this as we will always be using the existing tax lines - can you think of a case that wouldn't do this?

Reply:
for example if I create an order that is not placed? The alternative for the orders would be that if one of the items have tax_lines then we use them by default? and we remove the option